2-Operating-Systems/1-arch.md

```diff
@@ -0,0 +1,459 @@
+    echo '[[ -z $DISPLAY && ! -e /tmp/.X11-unix/X0 ]] && (( EUID )) && exec startx' \
+    > .bash_profile
+    exit
+    ```
+    
+    Explanation:
+    
+    ```
+    # Automatically login janedoe user on boot
+    mkdir -p /etc/systemd/system/getty@tty1.service.d
+    cat <<-EOF >     /etc/systemd/system/getty@tty1.service.d/override.conf
+        [Service]
+        ExecStart=
+        ExecStart=-/usr/bin/agetty --autologin janedoe --noclear %I     $TERM
+    EOF
+    
+    # Become your janedoe user
+    su - janedoe
+    
+    cat <<-EOF > .xinitrc
+    
+        #!/bin/bash
+    
+        # Let GPG know about our current terminal
+        gpg-connect-agent updatestartuptty /bye
+    
+        # Optional: Compositor for faster terminal rendering   
+        # compton &
+    
+        # Optional: Set wallpaper
+        # nitrogen --set-scaled ~/.wallpaper/yourcoolwallpaper.jpg
+    
+        # Optional: Start terminal
+        # kitty &
+    
+        # Optional: Set resolution and rotation
+        # xrandr --output HDMI1 --off --output DP1 --off --output eDP1 --mode 1200x1920 --pos 0x0 --rotate right --output VIRTUAL1 --off
+        # xinput set-prop 15 --type=float "Coordinate Transformation Matrix" 0 1 0 -1 0 1 0 0 1
+    
+        # Start Window manager
+        i3
+    EOF
+    
+    # Install default i3 config file
+    mkdir -p .config/i3
+    cp /etc/i3/config .config/i3/config
+    
+    # Optional: Set Window Manager font size
+    # sed -i "s/monospace .*/Terminus 15/g" .config/i3/config
+    
+    # Set default terminal
+    sed -i "s/i3-sensible-terminal/kitty/g" .config/i3/config
+    
+    # Configure media keys
+    cat <<-EOF >> .config/i3/config
+        bindsym XF86MonBrightnessUp exec xbacklight -inc 10
+        bindsym XF86MonBrightnessDown exec xbacklight -dec 10
+        bindsym XF86AudioRaiseVolume exec --no-startup-id pactl -- set-sink-volume 0 +5%
+        bindsym XF86AudioLowerVolume exec --no-startup-id pactl -- set-sink-volume 0 -5%
+        bindsym XF86AudioMute exec --no-startup-id pactl set-sink-mute 0 toggle  
+    EOF
+    
+    # Optional: Change default $mod key
+    # Mod4 = Windows/Mac/Meta key
+    # sed -i 's/Mod1/Mod4/g' .config/i3/config
+    
+    # Optional: Configure terminal and font
+    # mkdir -p .config/kitty
+    # echo "font_family Terminus" >> .config/kitty/kitty.conf
+    # echo "font_size 15.0" >> .config/kitty/kitty.conf
+    
+    # Optional: Scale browser for high resolution displays
+    # echo "--force-device-scale-factor=1.5" >> .config/chromium-flags.conf
+    
+    # Automatically start GUI environment if not already running
+    echo '[[ -z $DISPLAY && ! -e /tmp/.X11-unix/X0 ]] && (( EUID )) && exec startx' \
+    > .bash_profile
+    
+    exit
+    ```
+
+24. Start wifi on boot
+    
+    ```
+    # Get name of wireless device
+    ip addr # Usually starts with wl such as "wlp1s0"
+    systemctl enable netctl-auto@wlp1s05
```

Comment:
Either needs a sudo or a clarification that you should be root at this point. maybe do step 25 first as the user then escalate back to root then run this command or just add sudo here and on 26

Reply:
Why? systemctl should attempt elevation if you're not root.